frontend/src/components/dashboard/settings/Settings.tsx

```diff
@@ -100,6 +129,9 @@ const Settings = () => {
             <Route exact path="/settings/api">
               <ApiKey hasApiKey={hasApiKey} />
             </Route>
+            <Route exact path="/settings/email">
+              <CustomDomain onSuccess={fetchCustomFromAddresses} />
```

Comment:
Given that we call `getCustomFromAddresses` in this top-level component, perhaps we can pass the list of From Addresses down as a prop instead of calling it again in `populateFromAddresses` in the `CustomDomain` component. This will avoid holding state in the `CustomDomain` component.